fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -223,11 +223,21 @@ ThreadSafeDatabase::~ThreadSafeDatabase() {
 	onMainThreadVoid([db]() { db->delref(); });
 }
 
+ThreadSafeTenant::ThreadSafeTenant(Reference<ThreadSafeDatabase> db, StringRef name) : db(db), name(name) {
+	Tenant* tenant = this->tenant = Tenant::allocateOnForeignThread();
```

Comment:
Since tenant is a reference counted object and we aren't holding a `Reference` to it, I think we need to explicitly `delref` it when the `ThreadSafeTenant` is destructed.